digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -748,6 +753,99 @@ pub async fn get_token_supply(
     })
 }
 
+pub async fn get_token_accounts_by_owner(
+    conn: &impl ConnectionTrait,
+    owner_address: Vec<u8>,
+    mint_address: Option<Vec<u8>>,
+    token_program: Option<Vec<u8>>,
+) -> Result<SolanaRpcResponseAndContext<Vec<RpcData<RpcTokenInfo>>>, DbErr> {
+    let token_accounts = token_accounts::Entity::find();
+
+    let mut conditions = Condition::all().add(token_accounts::Column::Owner.eq(owner_address));
+
+    if let Some(ref mint) = mint_address {
+        conditions = conditions.add(token_accounts::Column::Mint.eq(mint.clone()));
+    }
+
+    if let Some(token_program) = token_program {
+        conditions = conditions.add(token_accounts::Column::TokenProgram.eq(token_program));
+    }
+
+    let token_accounts = token_accounts
+        .filter(conditions)
+        .order_by_desc(token_accounts::Column::Amount)
+        .all(conn)
+        .await?;
+
+    let mints = if let Some(mint) = mint_address {
+        vec![mint]
+    } else {
+        token_accounts
+            .iter()
+            .map(|ta| ta.mint.clone())
+            .collect::<Vec<_>>()
+    };
+
+    let mint_accounts = tokens::Entity::find()
+        .filter(tokens::Column::Mint.is_in(mints))
+        .all(conn)
+        .await?;
```

Comment:
Can this not be joined in to only query once?

We should look at doing similar thing to asset where have custom representation and fill with nulls if not supported.